lib/ansible/galaxy/role.py

```diff
@@ -194,17 +194,15 @@ def fetch(self, role_data):
         return False
 
     def install(self):
-        # the file is a tar, so open it that way and extract it
-        # to the specified (or default) roles directory
-        local_file = False
 
         if self.scm:
             # create tar file from scm url
-            tmp_file = RoleRequirement.scm_archive_role(**self.spec)
+            # in python 3.5 this cloud be (**self.options,**self.spec)
+            args = self.spec.copy()
+            args.update({"keep_scm_meta": self.options.keep_scm_meta})
+            tmp_file = RoleRequirement.scm_archive_role(**args)
```

Comment:
I suspect this will work even in Python-2.6:
``` python
tmp_file = RoleRequirement.scm_archive_role(keep_scm_meta=self.options.keep_scm_meta, **self.spec)
```

Reply:
Your right, I think I got caught up in wanting to pass in self.options originally. Updated